src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicInteropTests.cs

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Net.Security;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+using System.Reflection;
+using System.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+using Microsoft.Quic;
+
+namespace System.Net.Quic.Tests
+{
+    public class MsQuicInteropTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public MsQuicInteropTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        private static MemberInfo[] GetMembers<T>()
+        {
+            var members = typeof(T).FindMembers(MemberTypes.Field | MemberTypes.Property, BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Public, (mi, _) =>
+            {
+                if (mi is PropertyInfo property && property.GetSetMethod() == null)
+                {
+                    return false;
+                }
+
+                return true;
+            }, null);
+
+            Assert.NotEmpty(members);
+
+            return members;
+        }
+
+        private static void ResetMember(MemberInfo member, object instance)
+        {
+            switch (member)
+            {
+                case FieldInfo field:
+                    field.SetValue(instance, Activator.CreateInstance(field.FieldType));
+                    break;
+                case PropertyInfo property:
+                    property.SetValue(instance, Activator.CreateInstance(property.PropertyType));
+                    break;
+                default:
+                    throw new InvalidOperationException($"Unexpected member type: {member.MemberType}");
+            }
+        }
+
+
+        [Fact]
+        public void QuicSettings_Equals_RespectsAllMembers()
+        {
+            QUIC_SETTINGS settings = new QUIC_SETTINGS();
+
+            var settingsSpan = MemoryMarshal.AsBytes(new Span<QUIC_SETTINGS>(ref settings));
+
+            // Fill the memory with 1s,we will try to zero out each member and compare
+            settingsSpan.Fill(0xFF);
+
+            foreach (MemberInfo member in GetMembers<QUIC_SETTINGS>())
+            {
+                // copy and box the instance because reflection methods take a reference type arg
+                object boxed = settings;
```

Comment:
Will this copy the padding filling 1s as well?
Can we somehow construct the test (either this one or a separate one) in a way that will verify that the padding values are _not_ compared?

Idea for a separate test: fill with 1s, use reflection to fill all the fields with 0s, compare with `default` (should be equal despite paddings being 1s)

Reply:
we don't care about padding bytes anymore, I replaced the span-based comparison with listing all individual properties.

Reply:
Well, you replaced it for a _reason_ which is not currently stated anywhere in the code; so if in future someone (not aware of this discussion) will decide to bring the span-based comparison "back", they won't see any test failures.

It's like when you fix some bug but you still add a test to ensure the bug does not and will not reproduce anymore with your changes. It would be a kinda similar situation if we say "but we fixed the bug, so now we don't care about the previously broken scenario anymore"..?

That being said, this situation with padding feels like a corner case, and I guess it is unlikely that span-based comparison would be reintroduced. So I don't feel so strongly about that. More like trying to be extra cautious.

Reply:
> It's like when you fix some bug but you still add a test to ensure the bug does not (and will not) reproduce anymore with your changes.

I see tests more like a way to assert some assumptions about the behavior of the code. If we don't assume any behavior regarding padding bytes in the production (or test infra) code, then I don't see value of asserting such properties in tests. As opposed to when we find a bug and add a test asserting that in *that* scenario the code behaves the correct way.

If somebody decides to reintroduce the span-based approach, then I expect the question "is this safe/correct/sound?" to come up during code review (as it did on this PR).

I can add a comment to the implementation so that the reason for the decision is not lost.